### PR TITLE
Add support for logs volumes

### LINKED ISSUE #352: Add support for full-range histograms
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when exploring my logs because I cannot see the full range histograms.
Having full range histograms is really helpful when you need to understand a trend.
The current logs volume is based on the logs seen in the response which is not all logs in the range. Hence the logs volume is not really usable.

Capture d’écran 2024-03-22 à 09 14 03

**Describe the solution you'd like**
When is explore mode I would like to have support for full-range histograms for the logs volume.
I believe this feature can be achieve by splitting the API calls, one for the logs volume with the date histogram query and the second one for the logs themselves.

**Describe alternatives you've considered**
In explore you cannot workaround the lack of feature. However you can do it with kibana or graylog.
During my day to day I constantly switch back and forth between Grafana and Graylog, while I would love to be explore both my metrics and my logs all together in a single UI.

Example:
Capture d’écran 2024-03-22 à 09 24 49

**Additional context**
OpenSearch plugin: 2.14.4
Grafana: 10.1.8
OpenSearch: 1.3.14

### PR DESCRIPTION
**Pull request by psauvage0**
**What this PR does / why we need it**:

This add the LogsVolume feature (See https://grafana.com/developers/plugin-tools/tutorials/build-a-logs-data-source-plugin#show-full-range-logs-volume).

This allow the user to see all logs and not just those displayed in the Logs table view.

Here is a preview of the visualization in the Explore View :
Explore View

**Which issue(s) this PR fixes**:

Fixes #352 

**Special notes for your reviewer**:

Code is mostly adapted from ElasticSearch Plugin + Grafana Logs Datasource plugin documentation. Logs present in the screenshot are linux events from https://github.com/logpai/loghub 
Tests are also adapted from ElasticSearch Plugin, let us know if more tests are required

### CODE REVIEW

**Comment by jtheory** [comment]
They have 2 other possible PRs to add, but waiting to get feedback on this one -- 
happy also to help with regular maintenance on the plugin! but need confirmation

**Comment by taleena** [comment]
Hello @psauvage0 and thank you for the contribution! When looking at the documentation, I see that this API is still under development and a warning that this is not supported for external data sources. Opensearch is an external data source (not within the grafana/grafana repo), so I'm checking with our teams internally to see what other information and context is available.

**Comment by psauvage0** [comment]
Hi, thanks for your time @taleena :slightly_smiling_face: 

This feature is possible thanks to the work that has been done in this PR from last january : https://github.com/grafana/grafana/pull/80281. You might want to check with the people involved in it.
From what I saw in the PR, there should be no problem for external datasource anymore since the goal was precisely to decouple the internal datasources.
I'll stay available if you need more info on this.

**Comment by wilfriedroset** [comment]
As @jtheory said, we are planning to add support for Live tailing (like Loki) via https://github.com/grafana/opensearch-datasource/issues/299 and we want to add a highlight feature so that searched terms are highlighted

**Comment by matyax** [comment]
> This feature is possible thanks to the work that has been done in this PR from last january : https://github.com/grafana/grafana/pull/80281. You might want to check with the people involved in it.

Correct 👍

**Comment by taleena** [comment]
Thanks for the info here @psauvage0, and to @matyax for taking a look and confirming it's not an issue any longer, and now updating the docs!

The team owning this plugin can now review, we'll see if we can get to it next week. :)

File: src/opensearchDatasource.ts [hunk 1]
```diff
@@ -134,6 +145,102 @@ export class OpenSearchDatasource extends DataSourceWithBackend<OpenSearchQuery,
     this.sigV4Auth = settingsData.sigV4Auth ?? false;
   }
 
+  /**
+   * Private method used in the `getSupplementaryRequest` for DataSourceWithSupplementaryQueriesSupport, specifically for Logs volume queries.
+   * @returns An DataQueryRequest or undefined if no suitable queries are found.
```
Comment (kevinwcyu) [inline src/opensearchDatasource.ts:150]: minor nit:

```suggestion
   * @returns A DataQueryRequest or undefined if no suitable queries are found.
```

**Review by kevinwcyu — APPROVED** [review]
Nice work @psauvage0! Thanks for adding this feature. Just had some minor comments. I'll pre-approve it for now and will merge it once you commit the changes.

**Comment by psauvage0** [comment]
Hi @kevinwcyu and thanks for the review :slightly_smiling_face: 
I just pushed the commit for the fix

**Pull request merged**